Failed to `step` after Broken pipe error

Occasionally MineRL fails with `Broken pipe` error and never reconnect.

Here is my console log:

```
INFO     - 2019-06-19 20:57:18,227 - [chainerrl.experiments.train_agent train_agent 68] statistics:[('average_q', 0.3040304352121823), ('average_loss', 3.83408982771332), ('n_updates', 35494)]
ERROR    - 2019-06-19 20:58:09,334 - [minerl.env.malmo log_to_file 344] [20:58:09] [Client thread/INFO]: [STDOUT]: CLIENT request state: ERROR_TIMED_OUT_WAITING_FOR_EPISODE_PAUSE
ERROR    - 2019-06-19 20:58:09,335 - [minerl.env.malmo log_to_file 344] [20:58:09] [Client thread/INFO]: [STDOUT]: CLIENT enter state: ERROR_TIMED_OUT_WAITING_FOR_EPISODE_PAUSE
ERROR    - 2019-06-19 21:01:19,352 - [minerl.env.core _start_up 403] Failed to reset (timeout), trying again!
ERROR    - 2019-06-19 21:01:48,329 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: java.net.SocketException: Broken pipe (Write failed)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.net.SocketOutputStream.socketWrite0(Native Method)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.net.SocketOutputStream.write(SocketOutputStream.java:134)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.io.DataOutputStream.writeInt(DataOutputStream.java:198)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at com.microsoft.Malmo.Client.MalmoEnvServer.peek(MalmoEnvServer.java:591)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at com.microsoft.Malmo.Client.MalmoEnvServer.access$700(MalmoEnvServer.java:51)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDERR]: 	at com.microsoft.Malmo.Client.MalmoEnvServer$1.run(MalmoEnvServer.java:166)
ERROR    - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 344] [21:01:48] [EnvServerSocketHandler/INFO]: [STDOUT]: [ERROR] MalmoEnv socket error: java.net.SocketException: Broken pipe (Write failed) (can be on disconnect)
INFO     - 2019-06-19 21:01:48,330 - [minerl.env.malmo log_to_file 346] [21:01:48] [EnvServerSocketHandler/INFO]: [STDOUT]: [LOGTOPY] Want to quit on disconnect.
INFO     - 2019-06-19 21:01:50,522 - [chainerrl.experiments.train_agent save_agent 274] Saved the agent to results/MineRLTreechop-v0/rainbow_reset_rc300000_rs5000_gsoff_fs16/20190619T063940.084949/146976_except
ERROR    - 2019-06-19 21:01:50,522 - [__main__ main 257] execution failed.
Traceback (most recent call last):
  File "dqn_family.py", line 255, in main
    _main(args)
  File "dqn_family.py", line 383, in _main
    eval_env=eval_env,
  File "/usr/local/lib/python3.5/dist-packages/chainerrl/experiments/train_agent.py", line 163, in train_agent_with_evaluation
    logger=logger)
  File "/usr/local/lib/python3.5/dist-packages/chainerrl/experiments/train_agent.py", line 54, in train_agent
    obs, r, done, info = env.step(action)
  File "/usr/local/lib/python3.5/dist-packages/gym/core.py", line 282, in step
    return self.env.step(self.action(action))
  File "/usr/local/lib/python3.5/dist-packages/gym/core.py", line 258, in step
    observation, reward, done, info = self.env.step(action)
  File "/usr/local/lib/python3.5/dist-packages/gym/core.py", line 258, in step
    observation, reward, done, info = self.env.step(action)
  File "/usr/local/lib/python3.5/dist-packages/gym/core.py", line 258, in step
    observation, reward, done, info = self.env.step(action)
  File "/mnt/vol10/nakata/home/minerl2019/minerl-chainerrl/baselines/env_wrappers.py", line 37, in step
    obs, reward, done, info = self.env.step(action)
  File "/usr/local/lib/python3.5/dist-packages/gym/core.py", line 224, in step
    return self.env.step(action)
  File "/usr/local/lib/python3.5/dist-packages/chainerrl/wrappers/continuing_time_limit.py", line 38, in step
    observation, reward, done, info = self.env.step(action)
  File "/tmp/minerl2019/minerl/minerl/env/core.py", line 496, in step
    "Attempted to step an environment with done=True")
RuntimeError: Attempted to step an environment with done=True
ERROR    - 2019-06-19 21:01:50,550 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: java.io.EOFException
ERROR    - 2019-06-19 21:01:50,550 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.io.DataInputStream.readInt(DataInputStream.java:392)
ERROR    - 2019-06-19 21:01:50,551 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: 	at com.microsoft.Malmo.Client.MalmoEnvServer$1.run(MalmoEnvServer.java:143)
ERROR    - 2019-06-19 21:01:50,551 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: java.io.EOFException
ERROR    - 2019-06-19 21:01:50,551 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: 	at java.io.DataInputStream.readInt(DataInputStream.java:392)
ERROR    - 2019-06-19 21:01:50,551 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDERR]: 	at com.microsoft.Malmo.Client.MalmoEnvServer$1.run(MalmoEnvServer.java:143)
ERROR    - 2019-06-19 21:01:50,551 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDOUT]: [ERROR] MalmoEnv socket error: java.io.EOFException (can be on disconnect)
ERROR    - 2019-06-19 21:01:50,552 - [minerl.env.malmo log_to_file 344] [21:01:50] [EnvServerSocketHandler/INFO]: [STDOUT]: [ERROR] MalmoEnv socket error: java.io.EOFException (can be on disconnect)
INFO     - 2019-06-19 21:01:50,552 - [minerl.env.malmo log_to_file 346] [21:01:50] [EnvServerSocketHandler/INFO]: [STDOUT]: [LOGTOPY] Want to quit on disconnect.
INFO     - 2019-06-19 21:01:53,556 - [minerl.env.malmo on_terminate 200] Minecraft process psutil.Process(pid=294, status='terminated') terminated with exit code 0
```

And 
[This](https://github.com/minerllabs/minerl/files/3313456/minecraft_proc_9000.log) is MineRL-side log.

It seems that the state of MineRL becomes strange, because Gym says `RuntimeError: Attempted to step an environment with done=True`, which never happen.

--- a/minerl/data/data_pipeline.py
+++ b/minerl/data/data_pipeline.py
@@ -174,7 +174,7 @@ class DataPipeline:
             # for arg1, arg2, arg3 in files:
             #     DataPipeline._load_data_pyfunc(arg1, arg2, arg3)
             #     break
-            map_promise = self.processing_pool.starmap_async(DataPipeline._load_data_pyfunc, files)
+            map_promise = self.processing_pool.starmap_async(DataPipeline._load_data_pyfunc, files, error_callback=None)
 
             # random_queue = PriorityQueue(maxsize=pool_size)
 
@@ -364,7 +364,6 @@ class DataPipeline:
                 if frame_num == max_frame_num:
                     frames[-1] = frames[-2]
 
-
                 stop_idx = start_idx + len(frames)
                 # print('Num frames in batch:', stop_idx - start_idx)
 
@@ -399,7 +398,6 @@ class DataPipeline:
                 batches = [current_observation_data, action_data, [reward_data], next_observation_data, [np.array(done_data, dtype=np.bool)]]
                 if include_metadata:
                     batches += [meta]
-
 
                 if data_queue is None:
                     return batches

